containers/src/Kokkos_DualView.hpp

```diff
@@ -1099,6 +1088,33 @@ class DualView : public ViewTraits<DataType, Properties...> {
     }
   }
 
+  template <class... ViewCtorArgs>
+  inline void resync_host() {
+    using alloc_prop_input = Impl::ViewCtorProp<ViewCtorArgs...>;
+
+    // resync host mirror from device
+    if constexpr (alloc_prop_input::initialize) {
+      h_view = create_mirror_view(typename t_host::memory_space(), d_view);
+    } else {
+      h_view = create_mirror_view(Kokkos::WithoutInitializing,
+                                  typename t_host::memory_space(), d_view);
+    }
+  }
```

Comment:
```suggestion
  template <class... ViewCtorArgs>
  void resync_host(Impl::ViewCtorProp<ViewCtorArgs> const&) {
    using alloc_prop_input = Impl::ViewCtorProp<ViewCtorArgs...>;

    // resync host mirror from device
    if constexpr (alloc_prop_input::initialize) {
      h_view = create_mirror_view(typename t_host::memory_space(), d_view);
    } else {
      h_view = create_mirror_view(Kokkos::WithoutInitializing,
                                  typename t_host::memory_space(), d_view);
    }
  }
```

I would favor template deduction even though the input parameter is not used.

Can you also ensure this function is not public ?